src/univariate/continuous/cosine.jl

```diff
@@ -93,3 +93,27 @@ function ccdf(d::Cosine{T}, x::Real) where T<:Real
 end
 
 quantile(d::Cosine, p::Real) = quantile_bisect(d, p)
+
+function mgf(d::Cosine{T}, t::Real) where T<:Real
+    μ, σ = params(d)
+    t ≈ 0. && return exp(μ*t)
+    return T(π^2) * exp(μ*t) * sinh(σ*t) / (σ*t*(T(π^2) + (σ*t)^2))
```

Comment:
AFAICT the problem is not so much `t` in isolation but computing `sinh(σ*t) / (σ*t)` for small values of `σ*t`. Moreover, it seems a bit easier to cancel `π^2` and use the `IrrationalConstants.invπ` constant here. 

So I'd suggest something like

```suggestion
    σt = σ * t
    z = iszero(σt) ? one(float(σt)) : sinh(σ*t)/(σ*t)
    return exp(μ*t) * (z / (1 + (invπ * σt)^2))
```